script/proto_generate.sh

```diff
@@ -0,0 +1,19 @@
+git --version 2>&1 >/dev/null
+GIT_IS_AVAILABLE=$?
+if [ $GIT_IS_AVAILABLE -eq 0 ]
+then
```

Comment:
You don't need this if / then / then logic.  You can just check both dependencies at once.  Something along the lines of:

```
GIT_IS_AVAILABLE=$(git --version 2>&1 >/dev/null)
PROTOC_IS_AVAILABLE=$(protoc --version 2>&1 >/dev/null)

if $GIT_IS_AVAILABLE && $PROTOC_IS_AVAILABLE
then
    PERFORM LOGIC
else
    ERROR MESSAGING FOR BOTH GIT AND PROTOC NEED TO BE INSTALLED
fi


```